src/librustdoc/json/types.rs

```diff
@@ -0,0 +1,485 @@
+    pub paths: FxHashMap<Id, ItemSummary>,
+    /// Maps `crate_num` of items to a crate name and html_root_url if it exists
+    pub external_crates: FxHashMap<u32, ExternalCrate>,
+    /// A single version number to be used in the future when making backwards incompatible changes
+    /// to the JSON output.
+    pub format_version: u32,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ExternalCrate {
+    pub name: String,
+    pub html_root_url: Option<String>,
+}
+
+/// For external items (stuff not defined in the local crate), you don't get the same level of
+/// information. This struct should contain enough to generate a link/reference to the item in
+/// question, or can be used by a tool that takes the json output of multiple crates to find
+/// the actual item definition with all the relevant info.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ItemSummary {
+    pub crate_num: u32,
+    pub path: Vec<String>,
+    pub kind: ItemKind,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Item {
+    /// This can be used as a key to the `external_crates` map of [`Crate`][] to see which crate
+    /// this item came from.
+    pub crate_num: u32,
+    /// Some items such as impls don't have names.
+    pub name: Option<String>,
+    /// The source location of this item. May not be present if it came from a macro expansion,
+    /// inline assembly, other "virtual" files.
+    pub source: Option<Span>,
+    /// Usually documented items are all public, but you can tell rustdoc to output private items
+    /// so this field is needed to differentiate.
+    pub visibility: Visibility,
+    /// The full docstring of this item.
+    pub docs: String,
+    /// This mapping resolves [intradoc links](https://github.com/rust-lang/rfcs/blob/master/text/1946-intra-rustdoc-links.md) from the docstring to their IDs
+    pub links: FxHashMap<String, Id>,
+    /// Stringified versions of the attributes on this item (e.g. `"#[inline]"`)
+    pub attrs: Vec<String>,
+    pub deprecation: Option<Deprecation>,
+    pub kind: ItemKind,
+    pub inner: ItemEnum,
+    // TODO: should we stringify the cfg attrs as well, or should we preserve their structure so
+    // the consumer doesn't have to parse an arbitrarily nested tree to figure out what platforms
+    // the item is available on?
+    // TODO: should we have a "stability" field if it's only used by the standard library?
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Span {
+    /// The path to the source file for this span relative to the crate root.
+    pub filename: PathBuf,
+    /// Zero indexed Line and Column of the first character of the `Span`
+    pub begin: (usize, usize),
+    /// Zero indexed Line and Column of the last character of the `Span`
+    pub end: (usize, usize),
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Deprecation {
+    pub since: Option<String>,
```

Comment:
Nope, from [the edition guide](https://doc.rust-lang.org/edition-guide/rust-2018/the-compiler/an-attribute-for-deprecation.html): since can be in the future; you can put whatever you'd like, and what's put in there isn't checked.